File: ai-code_walkthrough/_filesets/ts/travel.ts
import axios from 'axios';

export async function timeTravel(apiKey: string, baseUrl: string, timelineId: string) {
  const response = await axios.post(`${baseUrl}/travel`, {
    destination: timelineId
  }, {
    headers: {
      'Authorization': `Bearer ${apiKey}`
    }
  });

  return response.data;
}

export async function registerItem(apiKey: string, baseUrl: string, timelineId: string) {
  const response = await axios.post(`${baseUrl}/items`, {
    timeline_id: timelineId,
    item_description: "Tesla's Blueprint",
    location: "Tesla's Lab"
  }, {
    headers: {
      'Authorization': `Bearer ${apiKey}`
    }
  });

  return response.data;
}

export async function checkParadox(apiKey: string, baseUrl: string, itemId: string) {
  const response = await axios.post(`${baseUrl}/paradox`, {
    item_id: itemId
  }, {
    headers: {
      'Authorization': `Bearer ${apiKey}`
    }
  });

  return response.data;
}

File: ai-code_walkthrough/_filesets/py/travel.py
import requests

def time_travel(api_key, base_url, timeline_id):
    response = requests.post(f"{base_url}/travel", json={
        "destination": timeline_id
    }, headers={
        'Authorization': f'Bearer {api_key}'
    })

    return response.json()

def register_item(api_key, base_url, timeline_id):
    response = requests.post(f"{base_url}/items", json={
        "timeline_id": timeline_id,
        "item_description": "Tesla's Blueprint",
        "location": "Tesla's Lab"
    }, headers={
        'Authorization': f'Bearer {api_key}'
    })

    return response.json()

def check_paradox(api_key, base_url, item_id):
    response = requests.post(f"{base_url}/paradox", json={
        "item_id": item_id
    }, headers={
        'Authorization': f'Bearer {api_key}'
    })

    return response.json()
